rust_src/src/casefiddle.rs

```diff
@@ -4,16 +4,46 @@ use std::ffi::CString;
 use remacs_macros::lisp_fn;
 
 use keymap::Ctl;
+use libc::ptrdiff_t;
 use lisp::defsubr;
 use lisp::LispObject;
 use lists::put;
-use remacs_sys::{case_action, casify_object, casify_region_nil, casify_word};
+use remacs_sys::{case_action, casify_object, casify_region_nil};
+use remacs_sys::{casify_region, scan_words};
 use remacs_sys::{control_x_map, initial_define_key, meta_map};
 use remacs_sys::{Qdisabled, Qnil, Qt};
+use threads::ThreadState;
 
+use editfns::goto_char;
 use obarray::intern;
 use symbols::symbol_value;
 
+fn casify_word(flag: case_action, object: LispObject) -> LispObject {
+    let words = object.as_fixnum_or_error();
+    let buffer_ref = ThreadState::current_buffer();
+
+    let far_end: ptrdiff_t = unsafe { scan_words(buffer_ref.pt, words) };
+    let far_end = match far_end {
```

Comment:
Maybe you shouldn't shadow this variable and make it mutable instead.

Reply:
Personally I prefer not to make variables mutable unless it's really necessary. Maybe I could move the function call inside the match clause if you prefer that?